src/components/SwitchContainer.ts

```diff
@@ -152,17 +159,31 @@ class SwitchContainer extends Component<SwitchContainerProps, SwitchContainerSta
         }
     }
 
-    private executeAction(actionname: string, guid: string) {
-        if (actionname && this.props.mxObject) {
-            window.mx.ui.action(actionname, {
-                error: (error) =>
-                    window.mx.ui.error(`Error while executing microflow ${actionname}: ${error.message}`),
+    private executeAction(mxObject: mendix.lib.MxObject) {
+        const { onChangeMicroflow, onChangeNanoflow, mxform } = this.props;
+
+        if (onChangeMicroflow) {
+            window.mx.ui.action(onChangeMicroflow, {
+                error: error =>
+                    window.mx.ui.error(`Error while executing microflow ${onChangeMicroflow}: ${error.message}`),
                 params: {
                     applyto: "selection",
-                    guids: [ guid ]
+                    guids: [ mxObject.getGuid() ]
```

Comment:
Missing origin fix for closing a popup on click of the widget
` origin: mxform`